pyspectrafuse/commands/build_cluster_db.py

```
@@ -0,0 +1,450 @@
+                ORDER BY m.cluster_id
+            ) TO '{psm_path}' (FORMAT PARQUET, COMPRESSION ZSTD)
+        """)
+
+        n_psms = conn.execute(f"SELECT COUNT(*) FROM read_parquet('{psm_path}')").fetchone()[0]
+        logger.info(f"PSM membership: {n_psms} rows written to {psm_path}")
+
+        # ── Cluster Stats + Purity via DuckDB ──
+        stats_purity = compute_stats_and_purity(psm_path)
+
+        # ── Best PSM per cluster + spectrum arrays (chunked to avoid OOM) ──
+        logger.info(f"Pass 2: Finding best PSM per cluster...")
```

Ruff (0.15.10)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)